packages/discovery-provider/src/tasks/entity_manager/entities/grant.py

```diff
@@ -90,40 +106,70 @@ def validate_grant_tx(params: ManageEntityParameters, metadata):
             raise IndexingValidationError(
                 f"Invalid Grant transaction, developer app address {metadata['grantee_address']} is invalid"
             )
+        if (
```

Comment:
Feels like we could stand to refactor this later to split up the developer app and user cases into separate functions just to make this a little less complicated.